amazon-qldb-driver-core/src/driver.rs

```diff
@@ -60,19 +61,32 @@ impl QldbDriverBuilder {
     /// Build a `QldbDriver` using the AWS SDK for Rust.
     ///
     /// `sdk_config` will be used to create a configured instance of the SDK.
-    pub async fn sdk_config(self, sdk_config: Config) -> QldbResult<QldbDriver<QldbSessionSdk>> {
-        let hyper = Client::https();
-        let client = QldbSessionSdk::new(hyper, sdk_config);
+    /// Note that this configuration uses the "sdk config" as opposed to the
+    /// QLDB specific config.
+    pub async fn sdk_config(
+        self,
+        sdk_config: &aws_types::config::Config,
+    ) -> QldbResult<QldbDriver<DynConnector>> {
+        let client = Client::new(sdk_config);
         self.build_with_client(client).await
     }
 
-    pub async fn build_with_client<C>(self, client: C) -> QldbResult<QldbDriver<C>>
+    /// Builds a `QldbDriver` using the AWS SDK for Rust.
+    ///
+    /// Note that `config` is the service-specific (QldbSession) config. For
+    /// shared config, see [`sdk_config`].
+    pub async fn config(self, config: Config) -> QldbResult<QldbDriver<DynConnector>> {
+        let client = Client::from_conf(config);
+        self.build_with_client(client).await
+    }
+
+    pub async fn build_with_client<C>(self, client: Client<C>) -> QldbResult<QldbDriver<C>>
     where
-        C: QldbSession + Send + Sync + Clone + 'static,
+        C: SmithyConnector,
     {
-        let ledger_name = self.ledger_name.ok_or(QldbError::UsageError(
-            "ledger_name must be initialized".to_string(),
-        ))?;
+        let ledger_name = self
+            .ledger_name
+            .ok_or(error::usage_error("ledger_name must be initialized"))?;
```

Comment:
If the `None` variant is illegal, why use an `Option` at all?

Reply:
This is a standard idiom with builders. Otherwise you need to make loads of builder structs to model a finite state machine:

```
struct QldbDriverBuilder;

impl QldbDriverBuilder {
  fn new() -> QldbDriverBuilder { QldbDriverBuilder {} }
  fn ledger_name(ledger_name: impl Into<String>) -> QldbDriverBuilderStep1 {
    QldbDriverBuilderStep1 { ledger_name: ledger_name.into() }
  }
}

struct QldbDriverBuilderStep1 {
  ledger_name: String
}

// ...
```

Because this isn't a "true runtime" error (i.e. something that randomly happens), it seems OK to allow for some misuse to escape the compiler.